gradio/chat_interface.py

```diff
@@ -746,3 +748,9 @@ async def _undo_msg(
         history, msg = await self._delete_prev_fn(msg, history)
         previous_msg = previous_input[-1] if len(previous_input) else msg
         return history, msg, previous_msg, previous_input
+
+    def render(self) -> None:
+        # If this is being rendered inside another Blocks, and the height is not explicitly set, set it to 400 instead of 200.
+        if get_blocks_context() and not self.provided_chatbot:
+            self.chatbot.height = 400
```

Comment:
Why do we set it to 400 instead of letting it fill the whole height? The discrepancy is the cause of the issues that people are seeing in https://github.com/gradio-app/gradio/issues/7714 and https://github.com/gradio-app/gradio/issues/7989

Reply:
This is possible with Blocks already if they set `gr.Blocks(fill_height=True)`. Otherwise, the Chatbot will take up 400 px.